crates/install-wheel-rs/src/uninstall.rs

```diff
@@ -208,6 +209,58 @@ pub fn uninstall_egg(egg_info: &Path) -> Result<Uninstall, Error> {
     })
 }
 
+/// Uninstall the legacy editable represented by the `.egg-link` file.
+///
+/// See: <https://github.com/pypa/pip/blob/41587f5e0017bcd849f42b314dc8a34a7db75621/src/pip/_internal/req/req_uninstall.py#L534-L552>
+pub fn uninstall_legacy_editable(egg_link: &Path) -> Result<Uninstall, Error> {
+    let mut file_count = 0usize;
+
+    let contents = fs::read_to_string(egg_link)?;
+
+    let target_line = if let Some(line) = contents.lines().find(|line| !line.is_empty()) {
+        line.trim()
+    } else {
+        return Err(Error::InvalidEggLink(egg_link.to_path_buf()));
+    };
+
+    match fs::remove_file(egg_link) {
+        Ok(()) => {
+            debug!("Removed file: {}", egg_link.display());
+            file_count += 1;
+        }
+        Err(err) if err.kind() == std::io::ErrorKind::NotFound => {}
+        Err(err) => return Err(err.into()),
+    }
+
+    let site_package = egg_link.parent().ok_or(Error::BrokenVenv(
+        "egg-link file is not in a directory".to_string(),
+    ))?;
+    let easy_install = site_package.join("easy-install.pth");
+
+    // Note concurrent use of `easy-install.pth` may result in lost writes
```

Comment:
This is my only concern. We might want to use an advisory file-lock on `easy-install.pth`? Or, actually, we could even use a mutex? Since we already have a lock on the environment itself, so in theory we only need to lock within this process.